gluefactory/eval/utils.py

```diff
@@ -69,6 +69,44 @@ def eval_matches_epipolar(data: dict, pred: dict) -> dict:
     return results
 
 
+def eval_matches_epipolar_via_gt_points(data: dict, pred: dict, conf) -> dict:
+    check_keys_recursive(data, ["view0", "view1", "pts_0to1"])
+    check_keys_recursive(
+        pred, ["keypoints0", "keypoints1", "matches0", "matching_scores0"]
+    )
+
+    kp0, kp1 = pred["keypoints0"], pred["keypoints1"]
+    m0, scores0 = pred["matches0"], pred["matching_scores0"]
+    pts0, pts1, scores = get_matches_scores(kp0, kp1, m0, scores0)
+
+    results = {}
+
+    estimator = load_estimator("fundamental_matrix", conf["estimator"])(conf)
+    data_ = {
+        "m_kpts0": pts0,
+        "m_kpts1": pts1,
+    }
+    est = estimator(data_)
+    success = est["success"] and (len(est["inliers"]) > 0)
+    if not success:
+        results["epi_error"] = [1e6 for i in range(len(data['pts_0to1']))]
+        results["ransac_inl"] = 0
+        results["ransac_inl%"] = 0
+    else:
+        M = est["M_0to1"]
+        inl = est["inliers"].numpy()
+        n_epi_err = sym_epipolar_distance(data['pts_0to1'][:,:2].double(), data['pts_0to1'][:,2:].double(), M.double(), squared=False).detach().cpu().numpy()
+        results["epi_error"] = n_epi_err
```

Comment:
results should be a float per entry (here it is a list). I suggest computing the AUC over the epipolar errors here, create one entry per threshold, and then just average them over all data points in the summary. This way it is compatible with the current setup.

Reply:
@Phil26AT unfortunately, that would result in a different number, because different images have very different number of the labeled keypoint correspondences. 
The way it is done now is in line with the paper, whereas one entry for threshold is not, I am afraid.